.travis.yml

```diff
@@ -34,15 +34,29 @@ matrix:
       after_failure:
         - docker-compose logs
     - language: generic
-      name: "StableX e2e Tests"
+      name: "StableX e2e Tests (Ganache)"
       cache: 
         directories:
           - dex-contracts/node_modules/
       install:
         - ./test/setup_contracts.sh
       script:
         - docker-compose up -d stablex
-        - ./test/e2e-tests-stablex.sh
+        - ./test/e2e-tests-stablex-ganache.sh
+      after_failure:
+        - docker-compose logs
+    - language: generic
+      name: "StableX e2e Tests (Rinkeby)"
+      cache: 
+        directories:
+          - dex-contracts/node_modules/
+      install:
+        - cd dex-contracts && npm ci && npx truffle compile && npm run networks-inject && cd ..
+        - echo -e $GITLAB_PRIVATE_KEY > .ssh/id_rsa && chmod 0500 .ssh/id_rsa
+        - docker-compose build --build-arg use_solver=1 stablex-rinkeby
+      script:
+        - docker-compose up -d stablex-rinkeby
+        - ./test/e2e-tests-stablex-rinkeby.sh
```

Comment:
Shouldn't this use `docker-compose run` so it uses docker? Otherwise it's kind of a mixed approach

Reply:
Interesting comment!

Reply:
It is kind of a mixed approach. The system under test runs in its target environment (a docker container), however the interactions with this environment are coming from the travis machine directly. We could run the script in another (separate) docker container, but I don't really see the value.

Reply:
I don't find it so important, so I wouldn't argue too much for this if it's already working, but a e2e test is built with all the pieces in docker, all pieces but one.

Docker deals for example with auto containing this node version, and any library you might depend on, so you can make sure it runs in travis or in any machine no matter the version they have for their global packages.

Reply:
Travis itself is containerized and does a clean install of all dependencies on each run. Is that what you mean by "auto containing"? Otherwise please explain, as I would like to understand this better.

Maybe you can explain a concrete scenario which shows the potential weakness of this approach.